src/modules/robot/Robot.cpp

```diff
@@ -639,8 +640,14 @@ void Robot::reset_axis_position(float x, float y, float z)
     this->transformed_last_milestone[Y_AXIS] = y;
     this->transformed_last_milestone[Z_AXIS] = z;
 
+    // check function pointer and call if set to transform the target to compensate for bed
+    if(compensationTransform) {
```

Comment:
FWIW I still think this is incorrect... homing did not use compensation transform, it set the actuators to a known position before compensation so both last_milestone and transformed_milestone are the same thing. *Homing does not use robot so there is no way for it to apply compensation). You can't then apply compensation transform without actually moving the head to match what you just set the actuators to.